pkg/apis/sessions/session_state.go

```diff
@@ -150,3 +147,50 @@ func into(s *string, f codecFunc) error {
 	*s = string(d)
 	return nil
 }
+
+// Compress & Decompress with LZ4
+// The Compress:Decompress ratio is 1:Many. LZ4 gives fastest decompress speeds
+func lz4Compress(payload []byte) ([]byte, error) {
+	buf := new(bytes.Buffer)
+	zw := lz4.NewWriter(nil)
+	zw.Header = lz4.Header{
+		BlockMaxSize:     65536,
+		CompressionLevel: 0,
+	}
+	zw.Reset(buf)
+
+	reader := bytes.NewReader(payload)
+	_, err := io.Copy(zw, reader)
+	if err != nil {
+		return nil, err
+	}
+	err = zw.Close()
+	if err != nil {
+		return nil, err
+	}
+
+	compressed, err := ioutil.ReadAll(buf)
+	if err != nil {
+		return nil, err
+	}
+
+	return compressed, nil
```

Comment:
Since you're not embellishing the error message with any extra detail, you could just `return ioutil.ReadAll(buf)`